### DISCUSSION TITLE
DynaTrace Plug-in Support

### DISCUSSION THREAD

**Discussion by jonathan-neufeld-asurion** [apache/pekko#505] [Q&A] [answered]
### DynaTrace Pekko Support
Does anyone know any details with regard to support for DynaTrace in Pekko?

### Background
DynaTrace is a monitoring tool that instruments Akka calls to collect telemetry forwarding to a collector node.

**Comment by mdedetrich** [comment]
I just quickly googled this and it looks like its a commercial product, so its up to them to support it. Considering that migrating from Akka 2.6 to Pekko 1.0.1 is largely just class/config/package renames it shouldn't be hard for them to support Pekko along with Akka.

I would raise the issue with them, we don't really have any control over this.

EDIT: Also in case its not clear, I highly doubt that their product would automatically work with Pekko, usually these telemetry based solutions work on injecting bytecode into certain method calls (i.e. Aspectj) and that is done by fully qualified method paths and since Pekko has changed all of the class/package names its no longer going to align.

**Reply by pjfanning** [reply, marked as answer]
I used to work for a company that used Dynatrace and we were just given an agent to install that monitored the JVMs. I think it has some heuristic approaches to discovering what to instrument. They could have certain logic in the agent that knows about certain classes to watch out for.

It would be best to talk to Dynatrace themselves.

**Reply by mdedetrich** [reply]
I also worked at a company that used Cinnamon which is a Ligthbend product that adds tracing to Akka, it also is set up as an agent but under the hood it uses https://en.wikipedia.org/wiki/AspectJ to implement the tracing
